test/WebJobs.Script.Tests/Description/DotNet/FunctionAssemblyLoadContextTests.cs

```diff
@@ -37,31 +39,21 @@ public void RuntimeAssemblies_AreLoadedInDefaultContext(string assemblyName)
             Assert.Same(AssemblyLoadContext.Default, AssemblyLoadContext.GetLoadContext(assembly));
         }
 
-        // TODO: FACAVAL
-        //[Fact]
-        //public void ResolveAssembly_WithIndirectPrivateDependency_LogsIfResolutionFails()
-        //{
-        //    TestLogger testLogger = new TestLogger("Test");
-
-        //    var resolver = new FunctionAssemblyLoader("c:\\");
-
-        //    var metadata1Directory = @"c:\testroot\test1";
-        //    var metadata1 = new FunctionMetadata { Name = "Test1", ScriptFile = $@"{metadata1Directory}\test.tst" };
-        //    var metadata2 = new FunctionMetadata { Name = "Test2", ScriptFile = @"c:\testroot\test2\test.tst" };
+        [Fact]
+        public void InitializeDeps_LoadsExpectedDependencies()
+        {
+            string depsPath = Path.Combine(Directory.GetCurrentDirectory(), @"Description\DotNet\TestFiles\DepsFiles");
 
-        //    var mockResolver = new Mock<IFunctionMetadataResolver>();
-        //    mockResolver.Setup(m => m.ResolveAssembly("MyTestAssembly.dll"))
-        //      .Returns<Assembly>(null);
+            IDictionary<string, string> assemblies = FunctionAssemblyLoadContext.InitializeDeps(depsPath);
 
-        //    resolver.CreateOrUpdateContext(metadata1, this.GetType().Assembly, new FunctionMetadataResolver(metadata1.ScriptFile, new Collection<ScriptBindingProvider>(), testLogger), testLogger);
-        //    resolver.CreateOrUpdateContext(metadata2, this.GetType().Assembly, mockResolver.Object, testLogger);
+            string testRid = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "win" : "unix";
 
-        //    Assembly result = resolver.ResolveAssembly(AppDomain.CurrentDomain, new System.ResolveEventArgs("MyTestAssembly.dll",
-        //        new TestAssembly(new AssemblyName("MyDirectReference"), @"file:///c:/testroot/test2/bin/MyDirectReference.dll")));
+            // Ensure runtime specific dependencies are resolved, with appropriate RID
+            Assert.Contains($"runtimes/{testRid}/lib/netstandard2.0/System.Private.ServiceModel.dll", assemblies.Values);
+            Assert.Contains($"runtimes/{testRid}/lib/netstandard1.3/System.Text.Encoding.CodePages.dll", assemblies.Values);
 
-        //    Assert.Null(result);
-        //    Assert.Equal(1, testLogger.GetLogMessages().Count);
-        //    Assert.Contains("MyTestAssembly.dll", testLogger.GetLogMessages()[0].FormattedMessage);
-        //}
+            // Ensure flattened dependency has expected path
+            Assert.Contains($"Microsoft.Azure.WebJobs.Host.Storage.dll", assemblies.Values);
```

Comment:
Sorry to keep asking clarifications :-) -- but this happens b/c this doesn't have any specified rid in the deps file?

Reply:
What happens? We have a non-RID specific dependency? Yes, this happens because the original package does not have runtime specific dependencies and the deps file reflects that.

If you're referring to the fact that they're flattened; the .NET SDK/build will flatten assembly paths, but will not do that for assemblies in the `runtimes` folder, hence the check.

Reply:
Sorry, wasn't clear in my question -- I was referring to the flattening. You answered me, though :-)